src/fsharp/CompileOps.fs

```diff
@@ -2205,12 +2205,7 @@ type TcConfigBuilder =
             | Some f,_ -> f
         let assemblyName, assemblyNameIsInvalid = 
             let baseName = fileNameOfPath outfile
-            let assemblyName = fileNameWithoutExtension baseName
-            if not (Filename.checkSuffix (String.lowercase baseName) (ext())) then
-                errorR(Error(FSComp.SR.buildMismatchOutputExtension(),rangeCmdArgs))
-                assemblyName, true
-            else
-                assemblyName, false
+            (fileNameWithoutExtension baseName), false
```

Comment:
`assemblyNameIsInvalid` is used only once [5 lines below](https://github.com/KevinRansom/visualfsharp/blob/5280e42ce16672d4b9eef188cc90d84025ec14eb/src/fsharp/CompileOps.fs#L2214)

```fsharp
if assemblyNameIsInvalid then None else
```

it can be removed, it's not used anymore